backend/endpoints/messages.py

```diff
@@ -204,3 +205,27 @@ async def update_message(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail={"message not edited": edited_message},
     )
+
+
+@router.get(
+    "/org/{org_id}/rooms/{room_id}/messages",
+    response_model=ResponseModel,
+    status_code=status.HTTP_200_OK,
+    responses={424: {"detail": "ZC Core failed"}},
+)
+async def get_messages(org_id, room_id):
+    """Fetches all messages sent in a particular room.
+
+    Args:
+        org_id (str): A unique identifier of an organization
+        room_id (str): A unique identifier of the room where messages are fetched from
+
+    Returns:
+        A list of message objects
+
+    """
+    room_messages = await get_room_messages(org_id, room_id)
+    return JSONResponse(
+        content=ResponseModel.success(data=room_messages, message="Messages retrieved"),
+        status_code=status.HTTP_200_OK,
+    )
```

Comment:
So, there's a utility function (get_room_messages) that helps to retrieve messages in a room as a list via DataStorage.
Here's the code:
DB = DataStorage(org_id)
    ```
response = await DB.read(MESSAGE_COLLECTION, query={"room_id": room_id})
    if response and "status_code" not in response:
        return response
    return {}
```
The issue is that, inside the endpoint, there's no way I can raise an exception (for example when Zc core fails). 
In fact, in the code above, the function returns either a list of messages or an empty dict.
When I test the get messages endpoint with a non-valid room, or a room in which there are no messages yet, I always get an empty dict. I guess I'll get the same result when Zc core will fail. If there are no messages, we should get an empty array, not an empty dict.
This is really ambiguous.

The main idea that comes into my mind is to refactor the utility function so that it will only return the raw response without any checking. Then, the necessary checks will be done in the endpoint function.

What do you think about this?

@mykie88 @pauline-banye @AI-fae @man-of-mind @kinglighthill

Reply:
@o4codes

Reply:
I think you can go ahead with the refactor.

Reply:
You can go ahead with the refactor @zxenonx , it's better that way.

Reply:
Ok. Thanks. I'll do that then.

Reply:
I think this problem will not be peculiar to this endpoint going forward, because most of those utility functions are returning empty dicts in all cases of a failed response or "falsy" return values.
IMO all possible response types (success, failed, falsy) should be accounted for in all the utility functions.